### PR TITLE
feat(my-account):  my account page layout and navigation

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**

### CODE REVIEW

File: components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  AccordionSummary,
+  Box,
+  Divider,
+  Typography,
+  useMediaQuery,
+  useTheme,
+  Link,
+} from '@mui/material'
+import { useTranslation } from 'next-i18next'
+import router from 'next/router'
```
Comment (kibo-kevinwatts) [inline components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx:18]: use 'useRouter' hook
import { useRouter } from 'next/router'

File: pages/my-account/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { MyAccountTemplate } from '@/components/page-templates'
+
+import type { GetServerSideProps, GetServerSidePropsContext, NextPage } from 'next'
+
+export const getServerSideProps: GetServerSideProps = async (
```
Comment (kibo-kevinwatts) [inline pages/my-account/index.tsx:7]: Need to secure this page, here in "getServerSideProps" like
https://nextjs.org/docs/authentication#authenticating-server-rendered-pages

or create nextJS middleware 
https://nextjs.org/docs/advanced-features/middleware

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx]: Resolved

File: pages/my-account/index.tsx (hunk 2 above)
Comment (kibo-geetanshuchhabra) [reply pages/my-account/index.tsx:7]: Resolved

File: components/page-templates/MyAccountTemplate/MyAccountTemplate.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    setup()
+    const fullWidthDividerComponent = screen.queryByTestId('full-width-divider-component')
+    const headingArray = screen.getAllByRole('heading', { level: 1 })
+    expect(fullWidthDividerComponent).not.toBeInTheDocument()
+    expect(headingArray[0]).toHaveTextContent('common:my-account')
+    expect(headingArray[1]).toHaveTextContent('order-details')
+  })
+
+  it('Mobile', async () => {
```
Comment (kibo-kevinwatts) [inline components/page-templates/MyAccountTemplate/MyAccountTemplate.spec.tsx:82]: what is being tested in these mobile/desktop tests?

if these tests are needed, update the names 'Desktop' or 'Mobile' is not enough detail

File: pages/_middleware.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  if (request.nextUrl.pathname.startsWith('/my-account')) {
+    if (!checkIsAutheticated(request)) {
+      return NextResponse.redirect('/')
+    }
+  }
+}
+
+export const config = {
+  matcher: '/my-account',
+}
```
Comment (kibo-kevinwatts) [inline pages/_middleware.ts:21]: nice, will this match other my-account paths? `/my-account/order-history` ?

**Review by kibo-kevinwatts — COMMENTED** [review]
this looks good to me, just not sure if those mobile / desktop tests are needed

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**